spring-integration-xml/src/test/java/org/springframework/integration/xml/config/XPathSplitterParserTests.java

```diff
@@ -63,11 +62,18 @@
 	SmartLifecycleRoleController roleController;
 
 	@Test
+	@EnabledOnJre(JRE.JAVA_8)
 	public void testXpathSplitterConfig() {
 		assertTrue(TestUtils.getPropertyValue(this.xpathSplitter, "createDocuments", Boolean.class));
 		assertFalse(TestUtils.getPropertyValue(this.xpathSplitter, "applySequence", Boolean.class));
 		assertFalse(TestUtils.getPropertyValue(this.xpathSplitter, "iterator", Boolean.class));
 		assertSame(this.outputProperties, TestUtils.getPropertyValue(this.xpathSplitter, "outputProperties"));
+		/*
+		 * JDK 10
+		 * WARNING: An illegal reflective access operation has occurred
+		 * WARNING: Illegal reflective access by org.springframework.util.ReflectionUtils
+		 * to field com.sun.org.apache.xpath.internal.jaxp.XPathExpressionImpl.xpath
```

Comment:
Interesting...
Maybe we can just end up with the `"xpathExpression.expression` path?

Anyway what is the premise of the error?
Is there any ideas ?

Thanks

Reply:
In a future compiler, such illegal accesses will be denied.

Reply:
In other words all our "Direct fields access" magic is going to be banned?